src/background/stats.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Ghostery Browser Extension
+ * https://www.ghostery.com/
+ *
+ * Copyright 2017-present Ghostery GmbH. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0
+ */
+import AnonymousCommunication from '@whotracksme/webextension-packages/packages/anonymous-communication';
+import Storage from '@whotracksme/webextension-packages/packages/anonymous-communication/storage';
+import {
+  getTimeAsYYYYMMDDHH,
+  getTimeAsYYYYMMDD,
+  getTrustedUtcTime,
+} from '@whotracksme/webextension-packages/packages/anonymous-communication/timestamps';
+
+const SERVER_URL = 'https://anonymous-communication.ghostery.net';
+const CONFIG_URL = 'https://api.ghostery.net/api/v1/config';
+const PUBLIC_KEYS_INDEXED_DB = 'anonymous-communication-public-keys';
+const STORAGE_KEY = 'anonymous-communication-stats';
+const HOUR = 1000 * 60 * 60;
+
+async function send(protocol) {
+  const now = getTrustedUtcTime();
+  const ts = getTimeAsYYYYMMDD(now);
+  const t = getTimeAsYYYYMMDDHH(now);
+
+  const data = await chrome.storage.local.get([STORAGE_KEY]);
+  const sentAt = Date.now();
```

Comment:
you can use `now` in here

Reply:
Yes and no :P the `now` is a date object, but it may change in the future (as it is an implementation detail of the library - it is written in the comment there). I wanted to avoid that, so I call for the timestamp directly.

Reply:
It might indeed change - for context, in the Ghostery Desktop Extension we have a mechanism to auto-correct the system clock - but you will have the guarantee that it will be a Unix epoche like Date.now. Also, note that is not theoretical; before we had it, we got data from users whose clock was constantly off. For Mobile, it seems less an issue though.

For the moment, we are not implementing something like that on Mobile, so it is only an abstraction. You are right that for managing the cooldown, you can use any clock that advances. Having a correct clock is more important for the timestamps in the message (the alive signal is not sensitive, but generally if you have a bad clock time it makes you stand out from the crowd). Still, I would recommend to only use the "trusted" time in the whole module, even though it will be currently not doing anything.

(Side-note: For robustness, skipping times that are far in the future is also useful: if you find a persisted value that is far from the future, dropping it protects against situations where the clock was once showing a date one year in the future. It might happen rarely, but if you run into the situation once that the clock jumped, the client might stop sending data for multiple years.)

Reply:
I recommend seeing the implementation of the trusted time :) it is nothing more than `return new Date()` ;)